### ISSUE TITLE
switch

### ISSUE THREAD

**Issue by felix-bohlin** [felix-bohlin/ui#21]
- [x] label
- [x] supporting text
- [x] validation

**Comment by hermit99** [comment]
First of all, fantastic job @felix-bohlin! ❤️ I'm probably one of the earliest adopters while it's still work in progress 😄 

Giving label a `role="switch"` triggers a11y warning: `Non-interactive element <label> cannot have interactive role 'switch`. Ref: https://svelte.dev/e/a11y_no_noninteractive_element_to_interactive_role.

Maybe remove `[role="switch"]` at line 5 of switch.css while set role="switch" to the `<input>` inside? May need to update all the examples in doc as well.

**Comment by felix-bohlin** [comment]
Hey @hermit99! Thank you 😊 Stoked to hear it!! There's a "Show and tell" section in the Discussion tab in this repo - I'd really like to see what you're up to at some point! 🙂

Looks like Svelte is more stringent with their a11y practices than [w3.org](https://www.w3.org/WAI/ARIA/apg/patterns/switch/) are in their examples. But that's great, I might start looking at how they're enforcing a11y to see if I can steal something off of them!

I'm on Christmas vacation atm without a computer, but you're welcome to start a PR if you want, that way you'll pop up as a contributor to the project 😊 And you're right, we'll need to change in the docs too. I know the Switch is used in the List component examples as well.

**Comment by hermit99** [comment]
Thanks @felix-bohlin. Wish you a Merry Christmas and wonderful vacation! 🎉 

PR #68 is now ready for review, but no rush, enjoy your holiday first!

**Comment by felix-bohlin** [comment]
Merged! Thanks again @hermit99!

**Comment by felix-bohlin** [comment]
Accordion to Github it can take up to 24 hours for a contribution credits to show up, so fingers crossed I didn't goof anything up by editing the PR before merging. 🙃
